### PR TITLE
Jupyterhub/Dask-Kubernetes setup on AWS

### PR DESCRIPTION
**Pull request by amanda-tan**
Using the setup for GCE but modified to run on AWS using the Heptio Quickstart Template.

### CODE REVIEW

**Comment by jhamman** [comment]
Thanks @atlehr. 

@rsignell-usgs and @jreadey - looking for thoughts / comments on this.

Nice work to all involved.

cc @yuvipanda and @mrocklin

File: aws/notebook/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    zict \
+    && conda clean -tipsy
+
+RUN jupyter labextension install @jupyter-widgets/jupyterlab-manager
+
+RUN pip install fusepy click jedi kubernetes --upgrade --no-cache-dir
+
+RUN pip install git+https://github.com/mrocklin/daskernetes@deploy \
+                git+https://github.com/zarr-developers/zarr \
+                git+https://github.com/jhamman/xarray@fix/zarr_set_attrs \
```
Comment (jhamman) [inline aws/notebook/Dockerfile]: No need to change immediately but this can be pydata/xarray@master now.

File: aws/notebook/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                git+https://github.com/martindurant/gcsfs.git@fuse \
+                git+https://github.com/dask/distributed@master \
+                git+https://github.com/dask/dask@master \
+                --no-cache-dir
```
Comment (jhamman) [inline aws/notebook/Dockerfile]: More generally, we can probably move to full releases / conda installs of a bunch of these.

File: aws/jupyter-config.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+proxy:
+  secretToken: "e1e761517a8f4c025144e53be60ffc482e6321e26d8a184c2bab289948b8ba13"
```
Comment (jreadey) [inline aws/jupyter-config.yaml]: Not a good idea to have secret checked in!

File: aws/notebook/.daskernetes.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        capabilities:
+          add: [SYS_ADMIN]
+        privileged: true
+      volume_mounts:
+        - mountPath: /dev/fuse
+          name: fuse
+      env:
+        - name: EXTRA_PIP_PACKAGES
+          value: git+https://github.com/dask/distributed
+        - name: GCSFUSE_BUCKET
```
Comment (jreadey) [inline aws/notebook/.daskernetes.yaml]: Does this translate to S3 bucket?  Use a more generic name for this environment variable?

File: aws/jupyter-config.yaml (hunk 3 above)
Comment (atlehr) [reply aws/jupyter-config.yaml]: Yep, should've checked before committing.

File: aws/notebook/Dockerfile (hunk 2 above)
Comment (rabernat) [reply aws/notebook/Dockerfile]: Is it possible to separate the python environment specification into its own file (environment.yml)? Or does something about docker require the whole DOCKERFILE to be a single self-contained script. Separating the python environment stuff from the more general setup could make it easier to define new environments down the line...

This is partly related to #87.

File: aws/notebook/.daskernetes.yaml (hunk 4 above)
Comment (martindurant) [reply aws/notebook/.daskernetes.yaml]: AWS can read from GCS, albeit less performantly, and I don't suppose there is a desire to duplicate the data. Having said that, there will be a need to transfer some form of credentials, if not on google hardware. 

There are s3/fuse solutions out there that, as far as I know, have not been tried in this context.

**Comment by jhamman** [comment]
Something that may be worth pursing here is to mount `s3://nasanex` via fuse. This would give us access to https://aws.amazon.com/public-datasets/nasa-nex/ and may be a nice, larger set of datasets for working on some of these file access issues.

File: aws/setup.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+helm init --service-account tiller
+kubectl --namespace=kube-system patch deployment tiller-deploy --type=json --patch='[{"op": "add", "path": "/spec/template/spec/containers/0/command", "value": ["/tiller", "--listen=localhost:44134"]}]'
+
+# Get Helm repositories
+helm repo add jupyterhub https://jupyterhub.github.io/helm-chart/
+helm repo update
+
+# Install JupyterHub and Dask on the cluster
+helm install jupyterhub/jupyterhub --version=v0.6.0-9701a90 --name=jupyter --namespace=pangeo -f jupyter-config.yaml
+helm install dask/dask --name=dask --namespace=pangeo -f dask-config.yaml
```
Comment (jhamman) [inline aws/setup.sh]: @mrocklin - question about `dask-config.yaml`. In the GCP deployment, you have also referenced this file but it is not included in the repository: 

https://github.com/pangeo-data/pangeo/blob/10dc105bae0083968417a8bec2b293ee1e6b4e2a/gce/setup.sh#L16-L18

Conversely, `jupyter-config.yaml` is included: https://github.com/pangeo-data/pangeo/blob/master/gce/jupyter-config.yaml

Any chance this was omitted from your commit/PR by accident?

Comment (mrocklin) [reply aws/setup.sh]: This file has been removed.  This line should be as well.

**Comment by jhamman** [comment]
After discussing with @amanda-tan / @robfatland - we've decided to close this PR for the time being. The http://pangeo-aws.cloudmaven.org/ jupyterhub will be down "for maintenance" for the foreseeable future. When bandwidth opens up on their end again, they may redeploy the cluster but the timetable for that is TBD. 

@rsignell-usgs and @jreadey are moving forward with AWS-pangeo developments.

**Pull request closed without merging**